src/PaymentGateways/PayPalCommerce/onBoardingRedirectHandler.php

```diff
@@ -0,0 +1,252 @@
+	 */
+	private function registerPayPalAccountConnectedNotice() {
+		Give_Admin_Settings::add_message( 'paypal-account-connected', esc_html__( 'PayPal account connected successfully.', 'give' ) );
+	}
+
+	/**
+	 * Register notice if Paypal error set in url.
+	 *
+	 * @since 2.8.0
+	 */
+	private function registerPayPalErrorNotice() {
+		Give_Admin_Settings::add_error( 'paypal-error', give_clean( $_GET['paypal-error'] ) );
+	}
+
+	/**
+	 * Return whether or not PayPal user redirect to GiveWP setting page after successful onboarding.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return bool
+	 */
+	private function isPayPalUserRedirected() {
+		return isset( $_GET['merchantIdInPayPal'] ) && Give_Admin_Settings::is_setting_page( 'gateways', 'paypal' );
+	}
+
+	/**
+	 * Return whether or not PayPal account details saved.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return bool
+	 */
+	private function isPayPalAccountDetailsSaved() {
+		return isset( $_GET['paypal-account-connected'] ) && Give_Admin_Settings::is_setting_page( 'gateways', 'paypal' );
+	}
+
+	/**
+	 * Return whether or not PayPal account details saved.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @return bool
+	 */
+	private function isPayPalError() {
+		return isset( $_GET['paypal-error'] ) && Give_Admin_Settings::is_setting_page( 'gateways', 'paypal' );
+	}
+
+
+	/**
+	 * validate rest api credential.
+	 *
+	 * @param  array  $array
+	 *
+	 * @since 2.8.0
+	 *
+	 */
+	private function didWeGetValidSellerRestApiCredentials( $array ) {
+
+		$required = [ 'client_id', 'client_secret' ];
+		$array    = array_filter( $array ); // Remove empty values.
+
+		if ( array_diff( $required, array_keys( $array ) ) ) {
+			$errorMessage = isset( $restApiCredentials['error_description'] ) ? urlencode( $restApiCredentials['error_description'] ) : '';
+			$this->redirectWhenOnBoardingFail( $errorMessage );
+		}
+	}
+
+	/**
+	 * Validate seller on Boarding status
+	 *
+	 * @param string $merchantId
+	 * @param string $accessToken
+	 *
+	 * @since 2.8.0
+	 *
+	 */
+	private function isAdminSuccessfullyOnBoarded( $merchantId, $accessToken ) {
+		$onBoardedData = (array) $this->getSellerOnBoardingDetailsFromPayPal( $merchantId, $accessToken );
+		$required      = [ 'payments_receivable', 'primary_email_confirmed' ];
+		$onBoardedData = array_filter( $onBoardedData ); // Remove empty values.
+
+		if ( array_diff( $required, array_keys( $onBoardedData ) ) ) {
+			$this->redirectWhenOnBoardingFail();
+		} elseif ( ! $onBoardedData['payments_receivable'] ) {
+			$this->redirectWhenOnBoardingFail( esc_html__( 'We are unable to connect your seller account because your seller account can not receive payments. Please contact PayPal Support Team', 'give' ) );
+		} elseif ( ! $onBoardedData['primary_email_confirmed'] ) {
+			$this->redirectWhenOnBoardingFail( esc_html__( 'We are unable to connect your seller account because your seller account primary email has not been confirmed. Please contact PayPal Support Team', 'give' ) );
+		}
```

Comment:
I don't _think_ we're saying that they didn't successfully connect, but that they need to do things on their account for it to be workably ready. I'm thinking we just append a list of needs something like:
```
Your are successfully connected, but you need to do a few things within your PayPal account before you're ready to receive donations:
- Confirm your primary email address
- Set up an account to receive payment from PayPal
```
Just have the missing items append to the list. That make sense?

Reply:
fixed: https://github.com/impress-org/givewp/compare/0e0e27168d4b...94c267311362

![image](https://user-images.githubusercontent.com/1784821/88331702-04809b80-cd4b-11ea-923e-92d846a9d6f6.png)